pkg/api/controllers/gitprovider/dto/dto.go

```diff
@@ -8,8 +8,10 @@ type RepositoryUrl struct {
 } // @name RepositoryUrl
 
 type SetGitProviderConfig struct {
-	Id         string  `json:"id" validate:"required"`
-	Username   *string `json:"username" validate:"optional"`
-	Token      string  `json:"token" validate:"required"`
-	BaseApiUrl *string `json:"baseApiUrl,omitempty" validate:"optional"`
+	Id            string  `json:"id" validate:"required"`
+	Username      *string `json:"username" validate:"optional"`
+	Token         string  `json:"token" validate:"required"`
+	BaseApiUrl    *string `json:"baseApiUrl,omitempty" validate:"optional"`
+	SigningKey    *string `json:"signingKey" validate:"optional"`
```

Comment:
Make sure to remove the key when it's being returned from the api. (similar to how `provider.Token = ""` is set in `ListGitProviders`)

Reply:
will do this after figure out why  dtn gp ls -f json both the fields are empty

Reply:
@bryans-go you skipped this

Reply:
so sorry